javav2/example_code/ecr/src/test/java/ECRTest.java

```diff
@@ -0,0 +1,122 @@
+import com.google.gson.Gson;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestMethodOrder;
+import software.amazon.awssdk.auth.credentials.EnvironmentVariableCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.ecr.EcrClient;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import com.example.ecr.scenario.ECRScenario;
+import software.amazon.awssdk.services.secretsmanager.SecretsManagerClient;
+import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueRequest;
+import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueResponse;
+
+@TestInstance(TestInstance.Lifecycle.PER_METHOD)
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public class ECRTest {
+
+    private static EcrClient ecrClient;
+
+    private static String repoName = "";
+
+    private static String newRepoName = "";
+    private static String iamRole = "" ;
+
+    private static ECRActions ecrActions;
+    @BeforeAll
+    public static void setUp() {
+        ecrClient = EcrClient.builder()
+            .region(Region.US_EAST_1)
+            .credentialsProvider(EnvironmentVariableCredentialsProvider.create())
+            .build();
+
+        ecrActions = new ECRActions();
+
+        // Get the values to run these tests from AWS Secrets Manager.
+        Gson gson = new Gson();
+        String json = getSecretValues();
+        SecretValues values = gson.fromJson(json, SecretValues.class);
+        newRepoName = values.getRepoName();
+        iamRole = values.getIamRole();
+        repoName = values.getExistingRepo();
+    }
+
+    @Test
+    @Tag("IntegrationTest")
+    @Order(1)
+    public void testScenario() {
+        assertDoesNotThrow(() -> ecrActions.createECRRepository(newRepoName));
+        assertDoesNotThrow(() -> ecrActions.setRepoPolicy(newRepoName, iamRole));
+        assertDoesNotThrow(() -> ecrActions.getRepoPolicy(newRepoName));
+        assertDoesNotThrow(() -> ecrActions.getAuthToken());
+        assertDoesNotThrow(() -> ecrActions.getRepositoryURI(newRepoName));
+        assertDoesNotThrow(() -> ecrActions.setLifeCyclePolicy(newRepoName));
+        assertDoesNotThrow(() -> ecrActions.pushDockerImage(newRepoName, newRepoName));
+        assertDoesNotThrow(() -> ecrActions.verifyImage(newRepoName, newRepoName));
+        assertDoesNotThrow(() -> ecrActions.deleteECRRepository(newRepoName));
+        System.out.println("Test 1 passed");
+    }
+
+    @Test
+    @Tag("IntegrationTest")
+    @Order(2)
+    public void testHello() {
+        assertDoesNotThrow(() -> HelloECR.listImageTags(ecrClient, repoName));
```

Comment:
The only way this test would throw an error is if the SDK ListImagesRequest.builder() failed. That doesn't seem to be testing much.

Reply:
Understood, but the only thing that _could_ fail in listImageTags is the builder. That seems like we're testing the SDK more than our code.